RiotNSE/NotificationService.swift

```diff
@@ -0,0 +1,499 @@
+        NSLog("[NotificationService] Payload came: \(userInfo)")
+        let roomId = userInfo["room_id"] as? String
+        let eventId = userInfo["event_id"] as? String
+
+        guard roomId != nil, eventId != nil else {
+            //  it's not a Matrix notification, do not change the content
+            NSLog("[NotificationService] Fallback case 7")
+            contentHandler(content)
+            return
+        }
+        
+        //  setup user account
+        setup()
+        
+        //  fetch the event first
+        fetchEvent()
+    }
+    
+    override func serviceExtensionTimeWillExpire() {
+        // Called just before the extension will be terminated by the system.
+        // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
+        NSLog("[NotificationService] Fallback case 5")
+        fallbackToOriginalContent()
+    }
+    
+    func setup() {
+        let sdkOptions = MXSDKOptions.sharedInstance()
+        sdkOptions.applicationGroupIdentifier = "group.im.vector"
+        sdkOptions.disableIdenticonUseForUserAvatar = true
+        sdkOptions.enableCryptoWhenStartingMXSession = true
+        sdkOptions.backgroundModeHandler = MXUIKitBackgroundModeHandler()
+        Bundle.mxk_customizeLocalizedStringTableName("Vector")
+        
+        if isatty(STDERR_FILENO) == 0 {
+            MXLogger.setSubLogName("nse")
+            MXLogger.redirectNSLog(toFiles: true)
+        }
+        
+        userAccount = MXKAccountManager.shared()?.activeAccounts.first
+        
+        if let userAccount = userAccount {
+            store = MXFileStore(credentials: userAccount.mxCredentials)
+            
+            if userAccount.mxSession == nil {
+                userAccount.openSession(with: store!)
+            }
+        }
+    }
+    
+    func fetchEvent() {
+        guard let content = originalContent, let userAccount = self.userAccount else {
+            //  there is something wrong, do not change the content
+            NSLog("[NotificationService] Fallback case 4")
+            fallbackToOriginalContent()
+            return
+        }
+        let userInfo = content.userInfo
+        
+        guard let roomId = userInfo["room_id"] as? String, let eventId = userInfo["event_id"] as? String else {
+            //  it's not a Matrix notification, do not change the content
+            NSLog("[NotificationService] Fallback case 1")
+            contentHandler?(content)
+            return
+        }
+        
+        userAccount.mxSession.event(withEventId: eventId, inRoom: roomId, success: { [weak self] (event) in
+            guard let self = self else {
+                NSLog("[NotificationService] Fallback case 9")
+                return
+            }
+            
+            guard let event = event else {
+                return
+            }
+            
+            guard event.isEncrypted else {
```

Comment:
```suggestion
            if !event.isEncrypted {
```

`guard` should be used for sanity check on values